hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/health/TestDeletingContainerHandler.java

```diff
@@ -0,0 +1,252 @@
+        .setContainerReplicas(containerReplicas)
+        .build();
+
+    Assert.assertTrue(deletingContainerHandler.handle(request));
+    Mockito.verify(replicationManager, Mockito.times(times))
+        .updateContainerState(Mockito.any(ContainerID.class),
+            Mockito.any(HddsProtos.LifeCycleEvent.class));
+  }
+
+  /**
+   * If a container is in Deleting state , some replicas exist and
+   * for each replica there is a pending delete, then do nothing.
+   */
+  @Test
+  public void testNoNeedResendDeleteCommand() throws NotLeaderException {
+    //ratis container
+    ContainerInfo containerInfo = ReplicationTestUtil.createContainerInfo(
+        ratisReplicationConfig, 1, DELETING);
+    Set<ContainerReplica> containerReplicas = ReplicationTestUtil
+        .createReplicas(containerInfo.containerID(),
+            ContainerReplicaProto.State.CLOSED, 0, 0, 0);
+    List<ContainerReplicaOp> pendingOps = new ArrayList<>();
+    containerReplicas.forEach(r -> pendingOps.add(
+        ContainerReplicaOp.create(ContainerReplicaOp.PendingOpType.DELETE,
+            r.getDatanodeDetails(), r.getReplicaIndex())));
+    resendDeleteCommand(containerInfo, containerReplicas, pendingOps, 0);
+
+    //EC container
+    containerInfo = ReplicationTestUtil.createContainerInfo(
+        ecReplicationConfig, 1, DELETING);
+    containerReplicas = ReplicationTestUtil
+        .createReplicas(containerInfo.containerID(),
+            ContainerReplicaProto.State.CLOSED, 1, 2, 3, 4, 5);
+    pendingOps.clear();
+    containerReplicas.forEach(r -> pendingOps.add(
+        ContainerReplicaOp.create(ContainerReplicaOp.PendingOpType.DELETE,
+            r.getDatanodeDetails(), r.getReplicaIndex())));
+    resendDeleteCommand(containerInfo, containerReplicas, pendingOps, 0);
+
+  }
+
+  /**
+   * If a container is in Deleting state , some replicas exist and
+   * for some replica there is no pending delete, then resending delete
+   * command.
+   */
+  @Test
+  public void testResendDeleteCommand() throws NotLeaderException {
+    //ratis container
+    ContainerInfo containerInfo = ReplicationTestUtil.createContainerInfo(
+        ratisReplicationConfig, 1, DELETING);
+    Set<ContainerReplica> containerReplicas = ReplicationTestUtil
+        .createReplicas(containerInfo.containerID(),
+            ContainerReplicaProto.State.CLOSED, 0, 0, 0);
+    List<ContainerReplicaOp> pendingOps = new ArrayList<>();
+    Set<ContainerReplica> tempContainerReplicas = new HashSet<>();
+    tempContainerReplicas.addAll(containerReplicas);
+    Iterator iter = tempContainerReplicas.iterator();
+    iter.next();
+    iter.remove();
+    Assert.assertEquals(2, tempContainerReplicas.size());
+    tempContainerReplicas.forEach(r -> pendingOps.add(
+        ContainerReplicaOp.create(ContainerReplicaOp.PendingOpType.DELETE,
+            r.getDatanodeDetails(), r.getReplicaIndex())));
+    resendDeleteCommand(containerInfo, containerReplicas, pendingOps, 1);
```

Comment:
This part is not very clear to me. Can you please explain what's being done here? 

If we're trying add the DELETE op to only two replicas, something like this could be easier to understand:
```suggestion
    containerReplicas.stream().limit(2).forEach(replica -> pendingOps.add(
        ContainerReplicaOp.create(ContainerReplicaOp.PendingOpType.DELETE,
            replica.getDatanodeDetails(), replica.getReplicaIndex())));
```